packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java

```diff
@@ -1058,6 +1078,10 @@ private void cancelDoKeyguardForChildProfilesLocked() {
         mDelayedProfileShowingSequence++;
     }
 
+    private void cancelDoRebootForOwnerAfterTimeoutIfEnabled() {
+        mDelayedRebootSequence++;
```

Comment:
can we also call `mAlarmManager.cancel` on the PendingIntent here? might need to refactor the PendingIntent creation from `doRebootForOwnerAfterTimeoutIfEnabled` so that a PendingIntent can be created for here too

Reply:
unsure how to do that. can you suggest something?

Reply:
Extract this part into a method that returns a PendingIntent:

```
private PendingIntent getRebootPendingIntent() {
    Intent rebootIntent = new Intent(DELAYED_REBOOT_ACTION);
    rebootIntent.putExtra("seq", mDelayedRebootSequence);
    rebootIntent.addFlags(Intent.FLAG_RECEIVER_FOREGROUND);
    return PendingIntent.getBroadcast(mContext, 0, rebootIntent, PendingIntent.FLAG_CANCEL_CURRENT);
}
```

Then, the `getRebootPendingIntent` method can be used in two places: to set the alarm and to cancel the alarm (https://developer.android.com/reference/android/app/AlarmManager#cancel(android.app.PendingIntent))

https://github.com/GrapheneOS/platform_frameworks_base/blob/bad07fdd0b4528ef166f0e9e692be641ac1ae99e/core/java/android/app/AlarmManager.java#L905-L915

https://github.com/GrapheneOS/platform_frameworks_base/blob/bad07fdd0b4528ef166f0e9e692be641ac1ae99e/core/java/android/content/Intent.java#L10331-L10360

Reply:
this would be something to consider for later